beatrix/src/test/java/org/killbill/billing/beatrix/integration/TestWithInArrearSubscriptions.java

```diff
@@ -150,6 +153,320 @@ public void testWithPauseResume() throws Exception {
+        assertEquals(bpEntitlement.getLastActivePlan().getName(), "premium-support-monthly-notrial");
+        checkNoMoreInvoiceToGenerate(account);
+
+        //CREATE ADDON SUBSCRIPTION. SINCE IN_ARREAR BILLING IS USED, INVOICE IS NOT GENERATED
+        final PlanPhaseSpecifier addOnSpec = new PlanPhaseSpecifier("premium-support-addon-monthly-notrial", null);
+        busHandler.pushExpectedEvents(NextEvent.CREATE, NextEvent.BLOCK, NextEvent.NULL_INVOICE);
+        final UUID addonEntitlementId = entitlementApi.addEntitlement(bpEntitlement.getBundleId(), new DefaultEntitlementSpecifier(addOnSpec), today, today, false, ImmutableList.<PluginProperty>of(), callContext);
+        Entitlement addOnEntitlement = entitlementApi.getEntitlementForId(addonEntitlementId, callContext);
+        assertEquals(addOnEntitlement.getState(), EntitlementState.ACTIVE);
+        assertEquals(addOnEntitlement.getLastActivePlan().getName(), "premium-support-addon-monthly-notrial");
+        assertListenerStatus();
+        checkNoMoreInvoiceToGenerate(account);
+
+        //MOVE CLOCK BY 1 MONTH
+        busHandler.pushExpectedEvents(NextEvent.INVOICE, NextEvent.PAYMENT, NextEvent.INVOICE_PAYMENT);
+        clock.addMonths(1);
+        assertListenerStatus();
+
+        invoiceChecker.checkInvoice(account.getId(), 1, callContext,
+                                    new ExpectedInvoiceItemCheck(new LocalDate(2012, 4, 1), new LocalDate(2012, 5, 1), InvoiceItemType.RECURRING, new BigDecimal("900.00")),
+                                    new ExpectedInvoiceItemCheck(new LocalDate(2012, 4, 1), new LocalDate(2012, 5, 1), InvoiceItemType.RECURRING, new BigDecimal("1000.00")));
+
+        //MOVE CLOCK
+        clock.addDays(10);
+
+        //CHANGE PLAN WITH START_OF_TERM POLICY
+        final PlanPhaseSpecifier newPlanSpec = new PlanPhaseSpecifier("basic-support-monthly-notrial", null);
+        busHandler.pushExpectedEvents(NextEvent.CREATE, NextEvent.CANCEL, NextEvent.BLOCK, NextEvent.NULL_INVOICE, NextEvent.NULL_INVOICE); //TODO: should this be CHANGE instead of CREATE? Also, need to check if other events are correct once bug is fixed
+        bpEntitlement.changePlanOverrideBillingPolicy(new DefaultEntitlementSpecifier(newPlanSpec), null, BillingActionPolicy.START_OF_TERM, null, callContext);
+        assertListenerStatus();
+        bpEntitlement = entitlementApi.getEntitlementForId(bpEntitlement.getId(), callContext);
+        assertEquals(bpEntitlement.getState(), EntitlementState.ACTIVE);
+        assertEquals(bpEntitlement.getLastActivePlan().getName(), "basic-support-monthly-notrial");
+        addOnEntitlement = entitlementApi.getEntitlementForId(addonEntitlementId, callContext);
+        assertEquals(addOnEntitlement.getState(), EntitlementState.CANCELLED);
+        checkNoMoreInvoiceToGenerate(account);
+
+    }
+
+    @Test(groups = "slow", enabled = false)
+    public void testChangeBPToPlanWithoutAddOnWithStartOfTermDate() throws Exception {
```

Comment:
And I suppose this one if for `https://github.com/killbill/killbill/issues/1631` (maybe add it to the `description =` annotation )field?